nautilus_trader/core/datetime.pyx

```diff
@@ -502,3 +502,71 @@ cpdef str format_iso8601_us(datetime dt):
 
     cdef tuple dt_partitioned = tz_stripped.rpartition('.')
     return f"{dt_partitioned[0]}.{dt_partitioned[2].rjust(6)}Z"
+
+
+cpdef int64_t iso8601_to_unix_millis(str iso8601):
+    """
+    Convert the given string to the Unix timestamp (microseconds).
+
+    Parameters
+    ----------
+    iso8601 : str
+        The input iso8601 datetime string to convert.
+
+    Notes
+    -----
+    Unit accuracy is millisecond.
+
+    Returns
+    -------
+    int64
+        The formatted string.
+
+    """
+    Condition.type(iso8601, str, "iso8601")
```

Comment:
The type `str` is already checked in the method signature

Reply:
Also, when returning primitive C types, we need to remember to add `except *` to the end of the method signature to ensure any exceptions are bubbled up to Python land. Otherwise C will swallow them near silently.

```
cpdef int64_t iso8601_to_unix_nanos(str iso8601) except *
```